src/school/repository/InMemorySchoolRepository.ts

```diff
@@ -4,6 +4,15 @@ import { SchoolRepository } from './SchoolRepository';
 export class InMemorySchoolRepository implements SchoolRepository {
   private _database = new Map<string, School>();
 
+  update(school: School): Promise<School> {
```

Comment:
Para hacer un update en memoria, no es necesario calcular un nuevo id, se debe reutilizar el que ya tiene la entidad school
```suggestion
 update(school: School): Promise<School> {
    this._database.set(`${school.id}`, school);
    return Promise.resolve();
  }
```